src/script_interface/get_value.hpp

```diff
@@ -96,6 +95,96 @@ template <typename T> T get_value(Variant const &v) {
   return detail::get_value_helper<T>{}(v);
 }
 
+/**
+ * @brief Get a value from a VariantMap by name, or throw
+ *        if it does not exist or is not convertiable to
+ *        the target type.
+ *
+ */
+template <typename T>
+T get_value(VariantMap const &vals, std::string const &name) {
+  try {
+    return get_value<T>(vals.at(name));
+  } catch (boost::bad_get const &) {
+    /* TODO: Better exceptions. */
+    throw;
+  } catch (std::out_of_range const &) {
+    /* TODO: Better exceptions. */
+    throw;
+  }
+}
+
+/**
+ * @brief Get a value from a VariantMap by name, or return a default
+ *        value if it does not exsist.
+ */
+template <typename T>
+T get_value_or(VariantMap const &vals, std::string const &name,
+               T const &default_) {
+  if (vals.count(name)) {
+    try {
+      return get_value<T>(vals.at(name));
+    } catch (std::out_of_range const &) {
+      /* TODO: Better exceptions. */
+      throw;
+    }
+  } else {
+    return default_;
+  }
+}
+
+/**
+ * @brief Make a new T with arguments extracted from a VariantMap.
+ */
+template <typename T, typename... Types, typename... ArgNames>
+T make_from_args(VariantMap const &vals, ArgNames... args) {
+  return T{get_value<Types>(vals, args)...};
+}
```

Comment:
```cpp
template <typename T, typename... Types, typename... ArgNames>
T make_from_args(VariantMap const &vals, ArgNames&&... args) {
  return T{get_value<Types>(vals, std::forward<ArgNames>(args))...};
}
```

Reply:
Yes, but in practice ArgNames are `const char *`, so there is not much gain here. You missed that `get_value` always forces a copy, which is a bigger isssue. Will fix both of course.